### PR TITLE
$a->get_baseurl() rewritten to App::get_baseurl() - 2nd PR

### PR DESCRIPTION
**Pull request by Quix0r**
I have created this PR because I had merged dbm::is_result() stuff into this branch which I wanted get merged into develop after this. #3027 is the original PR.

### CODE REVIEW

**Comment by Quix0r** [comment]
I have resolved it in my cherry-pick.

**Comment by Hypolite** [comment]
I don't exactly know what you wanted to do with this PR, but I spotted some `count()` -> `dbm::result()` changes in the diff.

**Comment by Quix0r** [comment]
Yes, they accidently went in before I started the previous PR. Okay, I can remove them here in this branch only.

**Comment by Quix0r** [comment]
Reverted more `count()` -> `dbm::is_result()`. That "don' cherry-pick" is a notice for myself. :-)

**Comment by Hypolite** [comment]
Next question: shouldn't we get rid of all the reference operators in function declarations where an `App` is passed as a parameter? Since it's an object, it is passed by reference anyway. Usage of reference operator is discouraged since PHP 5.

**Comment by Quix0r** [comment]
I would prefer it, else something like this is possible:

```
function bla (App &$a) {
$a = new App();
}
```

And now all further attempts on calling methods or using attributes from `$a` will be done on the new instance. This could lead to side-effects which you want to exclude. But I don't want to decide this here. Maybe more developers have to speak up on using & in method/function signatures.

**Comment by Quix0r** [comment]
I think usage of & should be sonsidered in a new PR as this is about get_baseurl() calls.

**Comment by Hypolite** [comment]
Fair enough, it already is a big PR.

File: include/salmon.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ function get_salmon_key($uri,$keyhash) {
 	// We have found at least one key URL
 	// If it's inline, parse it - otherwise get the key
 
-	if(count($ret)) {
+	if(count($ret) > 0) {
```
Comment (Hypolite) [inline include/salmon.php]: Standards: Could you please add a space after the `if`? 😊

File: include/security.php [hunk 2]
```diff
@@ -98,7 +98,7 @@ function authenticate_success($user_record, $login_initial = false, $interactive
 		call_hooks('logged_in', $a->user);
 
 		if(($a->module !== 'home') && isset($_SESSION['return_url']))
-			goaway($a->get_baseurl() . '/' . $_SESSION['return_url']);
+			goaway(App::get_baseurl() . '/' . $_SESSION['return_url']);
```
Comment (Hypolite) [inline include/security.php]: Standards: Could you please add brackets to this conditional statement? 😊

File: include/socgraph.php [hunk 3]
```diff
@@ -1181,22 +1182,23 @@ function update_suggestions() {
 	$done = array();
 
 	/// TODO Check if it is really neccessary to poll the own server
-	poco_load(0,0,0,$a->get_baseurl() . '/poco');
+	poco_load(0,0,0,App::get_baseurl() . '/poco');
 
-	$done[] = $a->get_baseurl() . '/poco';
+	$done[] = App::get_baseurl() . '/poco';
 
 	if(strlen(get_config('system','directory'))) {
```
Comment (Hypolite) [inline include/socgraph.php]: Standards: Could you please add a space after the `if`? 😊

File: include/socgraph.php [hunk 4]
```diff
@@ -1377,11 +1379,11 @@ function poco_discover_server($data, $default_generation = 0) {
 
 		if(isset($entry->urls)) {
 			foreach($entry->urls as $url) {
```
Comment (Hypolite) [inline include/socgraph.php]: Standards: Could you please add a space after the `if` and the `foreach`? 😊

File: index.php [hunk 5]
```diff
@@ -281,7 +281,7 @@
 
 		if((x($_SERVER,'QUERY_STRING')) && ($_SERVER['QUERY_STRING'] === 'q=internal_error.html') && isset($dreamhost_error_hack)) {
```
Comment (Hypolite) [inline index.php]: Standards: Could you please add a space after the `if`? 😊

File: index.php [hunk 6]
```diff
@@ -150,7 +150,7 @@
  *
  */
 
-// header('Link: <' . $a->get_baseurl() . '/amcd>; rel="acct-mgmt";');
+// header('Link: <' . App::get_baseurl() . '/amcd>; rel="acct-mgmt";');
 
 if(x($_COOKIE["Friendica"]) || (x($_SESSION,'authenticated')) || (x($_POST,'auth-params')) || ($a->module === 'login'))
```
Comment (Hypolite) [inline index.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/admin.php [hunk 7]
```diff
@@ -1131,13 +1131,13 @@ function admin_page_dbsync(&$a) {
 	}
 	if(! count($failed)) {
```
Comment (Hypolite) [inline mod/admin.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/admin.php [hunk 8]
```diff
@@ -1547,10 +1547,10 @@ function admin_page_plugins(&$a){
 	 */
 
 	if(x($_GET,"a") && $_GET['a']=="r") {
```
Comment (Hypolite) [inline mod/admin.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/allfriends.php [hunk 9]
```diff
@@ -26,7 +28,8 @@ function allfriends_content(&$a) {
 		intval(local_user())
 	);
 
-	if(! count($c))
+	if (! count($c)) {
+	}
```
Comment (Hypolite) [inline mod/allfriends.php]: The closing bracket should be moved after the following `return;` statement. Thanks for adding them though!

File: mod/cal.php [hunk 10]
```diff
@@ -231,8 +231,9 @@ function cal_content(&$a) {
 			$r = sort_by_date($r);
 			foreach($r as $rr) {
 				$j = (($rr['adjust']) ? datetime_convert('UTC',date_default_timezone_get(),$rr['start'], 'j') : datetime_convert('UTC','UTC',$rr['start'],'j'));
-				if(! x($links,$j))
-					$links[$j] = $a->get_baseurl() . '/' . $a->cmd . '#link-' . $j;
+				if(! x($links,$j)) {
```
Comment (Hypolite) [inline mod/cal.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/delegate.php [hunk 11]
```diff
@@ -38,21 +39,22 @@ function delegate_content(&$a) {
 				);
 			}
 		}
-		goaway($a->get_baseurl() . '/delegate');
+		goaway(App::get_baseurl() . '/delegate');
 	}
 
 	if($a->argc > 2 && $a->argv[1] === 'remove' && intval($a->argv[2])) {
```
Comment (Hypolite) [inline mod/delegate.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/dfrn_confirm.php [hunk 12]
```diff
@@ -504,7 +504,7 @@ function dfrn_confirm_post(&$a,$handsfree = null) {
 		// do anything special with this new friend.
 
 		if($handsfree === null)
-			goaway($a->get_baseurl() . '/contacts/' . intval($contact_id));
+			goaway(App::get_baseurl() . '/contacts/' . intval($contact_id));
 		else
 			return;
```
Comment (Hypolite) [inline mod/dfrn_confirm.php]: Standards: Could you please add brackets to this whole conditional statement? 😊

File: mod/dfrn_request.php [hunk 13]
```diff
@@ -501,13 +504,13 @@ function dfrn_request_post(&$a) {
 			else {
 				if(! validate_url($url)) {
```
Comment (Hypolite) [inline mod/dfrn_request.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/dfrn_request.php [hunk 14]
```diff
@@ -501,13 +504,13 @@ function dfrn_request_post(&$a) {
...
 					notice( t('Invalid profile URL.') . EOL);
-					goaway($a->get_baseurl() . '/' . $a->cmd);
+					goaway(App::get_baseurl() . '/' . $a->cmd);
 					return; // NOTREACHED
 				}
 
 				if(! allowed_url($url)) {
```
Comment (Hypolite) [inline mod/dfrn_request.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/dfrn_request.php [hunk 15]
```diff
@@ -518,7 +521,7 @@ function dfrn_request_post(&$a) {
 
 				if(! count($parms)) {
```
Comment (Hypolite) [inline mod/dfrn_request.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/dfrn_request.php [hunk 16]
```diff
@@ -518,7 +521,7 @@ function dfrn_request_post(&$a) {
...
 					notice( t('Profile location is not valid or does not contain profile information.') . EOL );
-					goaway($a->get_baseurl() . '/' . $a->cmd);
+					goaway(App::get_baseurl() . '/' . $a->cmd);
 				}
 				else {
 					if(! x($parms,'fn'))
```
Comment (Hypolite) [inline mod/dfrn_request.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/dfrn_request.php [hunk 17]
```diff
@@ -806,7 +812,7 @@ function dfrn_request_content(&$a) {
 			$myaddr = $_GET['address'];
 		elseif(local_user()) {
```
Comment (Hypolite) [inline mod/dfrn_request.php]: Standards: Can you please review this whole conditional structure for brackets and spaces? 😊

File: mod/dirfind.php [hunk 18]
```diff
@@ -197,7 +197,7 @@ function dirfind_content(&$a, $prefix = "") {
 					} else
```
Comment (Hypolite) [inline mod/dirfind.php]: Standards: Could you please add brackets to this conditional statement? 😊

File: mod/events.php [hunk 19]
```diff
@@ -9,8 +9,9 @@
 require_once('include/items.php');
 
 function events_init(&$a) {
-	if(! local_user())
+	if (! local_user()) {
 		return;
+	}
 
 	if($a->argc == 1) {
```
Comment (Hypolite) [inline mod/events.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/events.php [hunk 20]
```diff
@@ -336,8 +341,9 @@ function events_content(&$a) {
 			$r = sort_by_date($r);
 			foreach($r as $rr) {
```
Comment (Hypolite) [inline mod/events.php]: Standards: Could you please add a space after the `foreach`? 😊

File: mod/events.php [hunk 21]
```diff
@@ -336,8 +341,9 @@ function events_content(&$a) {
...
 				$j = (($rr['adjust']) ? datetime_convert('UTC',date_default_timezone_get(),$rr['start'], 'j') : datetime_convert('UTC','UTC',$rr['start'],'j'));
-				if(! x($links,$j))
-					$links[$j] = $a->get_baseurl() . '/' . $a->cmd . '#link-' . $j;
+				if(! x($links,$j)) {
```
Comment (Hypolite) [inline mod/events.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/filerm.php [hunk 22]
```diff
@@ -21,7 +21,7 @@ function filerm_content(&$a) {
 		file_tag_unsave_file(local_user(),$item_id,$term, $category);
 
 	if(x($_SESSION,'return_url'))
```
Comment (Hypolite) [inline mod/filerm.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/follow.php [hunk 23]
```diff
@@ -175,12 +175,12 @@ function follow_post(&$a) {
 			notice($result['message']);
 		goaway($return_url);
 	} elseif ($result['cid'])
```
Comment (Hypolite) [inline mod/follow.php]: Standards: Could you please add brackets to this conditional statement? 😊

File: mod/follow.php [hunk 24]
```diff
@@ -175,12 +175,12 @@ function follow_post(&$a) {
...
-		goaway($a->get_baseurl().'/contacts/'.$result['cid']);
+		goaway(App::get_baseurl().'/contacts/'.$result['cid']);
 
 	info( t('Contact added').EOL);
 
 	if(strstr($return_url,'contacts'))
```
Comment (Hypolite) [inline mod/follow.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/group.php [hunk 25]
```diff
@@ -28,36 +28,41 @@ function group_post(&$a) {
 		if($r) {
```
Comment (Hypolite) [inline mod/group.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/hcard.php [hunk 26]
```diff
@@ -38,14 +38,15 @@ function hcard_init(&$a) {
...
-	$a->page['htmlhead'] .= '<link rel="lrdd" type="application/xrd+xml" href="' . $a->get_baseurl() . '/xrd/?uri=' . $uri . '" />' . "\r\n";
-	header('Link: <' . $a->get_baseurl() . '/xrd/?uri=' . $uri . '>; rel="lrdd"; type="application/xrd+xml"', false);
-  	
+	$a->page['htmlhead'] .= '<link rel="lrdd" type="application/xrd+xml" href="' . App::get_baseurl() . '/xrd/?uri=' . $uri . '" />' . "\r\n";
+	header('Link: <' . App::get_baseurl() . '/xrd/?uri=' . $uri . '>; rel="lrdd"; type="application/xrd+xml"', false);
+
 	$dfrn_pages = array('request', 'confirm', 'notify', 'poll');
-	foreach($dfrn_pages as $dfrn)
-		$a->page['htmlhead'] .= "<link rel=\"dfrn-{$dfrn}\" href=\"".$a->get_baseurl()."/dfrn_{$dfrn}/{$which}\" />\r\n";
+	foreach($dfrn_pages as $dfrn) {
```
Comment (Hypolite) [inline mod/hcard.php]: Standards: Could you please add a space after the `foreach`? 😊

File: mod/install.php [hunk 27]
```diff
@@ -518,14 +518,13 @@ function check_smarty3(&$checks) {
 }
 
 function check_htaccess(&$checks) {
-	$a = get_app();
 	$status = true;
 	$help = "";
 	if (function_exists('curl_init')){
```
Comment (Hypolite) [inline mod/install.php]: Standards: Could you please add a space before the opening bracket? 😊

File: mod/install.php [hunk 28]
```diff
@@ -518,14 +518,13 @@ function check_smarty3(&$checks) {
...
-		$test = fetch_url($a->get_baseurl()."/install/testrewrite");
+		$test = fetch_url(App::get_baseurl()."/install/testrewrite");
 
 		if ($test!="ok")
```
Comment (Hypolite) [inline mod/install.php]: Standards: Could you please add brackets to this conditional statement? 😊

File: mod/invite.php [hunk 29]
```diff
@@ -119,7 +119,7 @@ function invite_content(&$a) {
 		if($a->config['register_policy'] == REGISTER_CLOSED)
 			$linktxt = sprintf( t('Visit %s for a list of public sites that you can join. Friendica members on other sites can all connect with each other, as well as with members of many other social networks.'), $dirloc . '/siteinfo');
 		elseif($a->config['register_policy'] != REGISTER_CLOSED)
```
Comment (Hypolite) [inline mod/invite.php]: Standards: Could you please review this whole conditional statement for brackets and spaces? 😊

File: mod/item.php [hunk 30]
```diff
@@ -62,7 +62,7 @@ function item_post(&$a) {
 	if(!$preview && x($_REQUEST['post_id_random'])) {
 		if(x($_SESSION['post-random']) && $_SESSION['post-random'] == $_REQUEST['post_id_random']) {
```
Comment (Hypolite) [inline mod/item.php]: Standards: Could you please add a space after the `if`s? 😊

File: mod/item.php [hunk 31]
```diff
@@ -449,7 +449,7 @@ function item_post(&$a) {
 			$objecttype = ACTIVITY_OBJ_IMAGE;
 
 			foreach($images as $image) {
```
Comment (Hypolite) [inline mod/item.php]: Standards: Could you please add a space after the `foreach`? 😊

File: mod/item.php [hunk 32]
```diff
@@ -449,7 +449,7 @@ function item_post(&$a) {
...
-				if(! stristr($image,$a->get_baseurl() . '/photo/'))
+				if(! stristr($image,App::get_baseurl() . '/photo/'))
```
Comment (Hypolite) [inline mod/item.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/item.php [hunk 33]
```diff
@@ -640,7 +641,7 @@ function item_post(&$a) {
 			if (dbm::is_result($r)) {
 				if(strlen($attachments))
```
Comment (Hypolite) [inline mod/item.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/noscrape.php [hunk 34]
```diff
@@ -26,6 +26,7 @@ function noscrape_init(&$a) {
 	$keywords = str_replace(array('#',',',' ',',,'),array('',' ',',',','),$keywords);
 	$keywords = explode(',', $keywords);
 
+	/// @TODO This query's result is not being used (see below), maybe old-lost code?
```
Comment (Hypolite) [inline mod/noscrape.php]: Actually it is used, see line 40 `'photo' => $r[0]["photo"],`

File: mod/noscrape.php [hunk 35]
```diff
@@ -59,12 +60,16 @@ function noscrape_init(&$a) {
 
 	//These are optional fields.
 	$profile_fields = array('pdesc', 'locality', 'region', 'postal-code', 'country-name', 'gender', 'marital', 'about');
-	foreach($profile_fields as $field)
-		if(!empty($a->profile[$field])) $json_info["$field"] = $a->profile[$field];
+	foreach($profile_fields as $field) {
+		if(!empty($a->profile[$field])) {
```
Comment (Hypolite) [inline mod/noscrape.php]: Standards: Could you please add a space after the `if` and the `foreach`? 😊

File: mod/noscrape.php [hunk 36]
```diff
@@ -59,12 +60,16 @@ function noscrape_init(&$a) {
...
+			$json_info["$field"] = $a->profile[$field];
+		}
+	}
 
 	$dfrn_pages = array('request', 'confirm', 'notify', 'poll');
-	foreach($dfrn_pages as $dfrn)
-		$json_info["dfrn-{$dfrn}"] = $a->get_baseurl()."/dfrn_{$dfrn}/{$which}";
+	foreach($dfrn_pages as $dfrn) {
```
Comment (Hypolite) [inline mod/noscrape.php]: Standards: Could you please add a space after the `foreach`? 😊

File: mod/profile.php [hunk 37]
```diff
@@ -56,15 +56,16 @@ function profile_init(&$a) {
...
-	$a->page['htmlhead'] .= '<link rel="lrdd" type="application/xrd+xml" href="' . $a->get_baseurl() . '/xrd/?uri=' . $uri . '" />' . "\r\n";
-	header('Link: <' . $a->get_baseurl() . '/xrd/?uri=' . $uri . '>; rel="lrdd"; type="application/xrd+xml"', false);
+	$a->page['htmlhead'] .= '<link rel="lrdd" type="application/xrd+xml" href="' . App::get_baseurl() . '/xrd/?uri=' . $uri . '" />' . "\r\n";
+	header('Link: <' . App::get_baseurl() . '/xrd/?uri=' . $uri . '>; rel="lrdd"; type="application/xrd+xml"', false);
 
 	$dfrn_pages = array('request', 'confirm', 'notify', 'poll');
-	foreach($dfrn_pages as $dfrn)
-		$a->page['htmlhead'] .= "<link rel=\"dfrn-{$dfrn}\" href=\"".$a->get_baseurl()."/dfrn_{$dfrn}/{$which}\" />\r\n";
-	$a->page['htmlhead'] .= "<link rel=\"dfrn-poco\" href=\"".$a->get_baseurl()."/poco/{$which}\" />\r\n";
+	foreach($dfrn_pages as $dfrn) {
```
Comment (Hypolite) [inline mod/profile.php]: Standards: Could you please add a space after the `foreach`? 😊

File: mod/settings.php [hunk 38]
```diff
@@ -628,8 +630,9 @@ function settings_post(&$a) {
 	if(($old_visibility != $net_publish) || ($page_flags != $old_page_flags)) {
```
Comment (Hypolite) [inline mod/settings.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/starred.php [hunk 39]
```diff
@@ -7,19 +7,23 @@ function starred_init(&$a) {
...
 	$r = q("SELECT starred FROM item WHERE uid = %d AND id = %d LIMIT 1",
 		intval(local_user()),
 		intval($message_id)
 	);
-	if(! dbm::is_result($r))
+	if (! dbm::is_result($r)) {
 		killme();
+	}
 
 	if(! intval($r[0]['starred']))
```
Comment (Hypolite) [inline mod/starred.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/starred.php [hunk 40]
```diff
@@ -39,7 +43,7 @@ function starred_init(&$a) {
 		if(strpos($return_path, '?')) $rand = "&$rand";
 		else $rand = "?$rand";
```
Comment (Hypolite) [inline mod/starred.php]: Standards: Could you please review this whole conditional statement for brackets and spaces? 😊

File: mod/subthread.php [hunk 41]
```diff
@@ -41,8 +41,9 @@ function subthread_content(&$a) {
 			intval($item['contact-id']),
 			intval($item['uid'])
 		);
-		if(! dbm::is_result($r))
+		if (! dbm::is_result($r)) {
 			return;
+		}
 		if(! $r[0]['self'])
```
Comment (Hypolite) [inline mod/subthread.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/tagger.php [hunk 42]
```diff
@@ -97,7 +97,7 @@ function tagger_content(&$a) {
 	if(! isset($bodyverb))
```
Comment (Hypolite) [inline mod/tagger.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: mod/tagrm.php [hunk 43]
```diff
@@ -51,30 +53,31 @@ function tagrm_content(&$a) {
...
 
-	if(! local_user()) {
-		goaway($a->get_baseurl() . '/' . $_SESSION['photo_return']);
+	if (! local_user()) {
+		goaway(App::get_baseurl() . '/' . $_SESSION['photo_return']);
 		// NOTREACHED
 	}
 
 	$item = (($a->argc > 1) ? intval($a->argv[1]) : 0);
 	if(! $item) {
```
Comment (Hypolite) [inline mod/tagrm.php]: Standards: Could you please add a space after the `if`? 😊

File: mod/xrd.php [hunk 44]
```diff
@@ -31,14 +32,14 @@ function xrd_init(&$a) {
...
 		'$dspr_guid' => $r[0]['guid'],
 		'$dspr_key' => base64_encode(pemtorsa($r[0]['pubkey']))
 	));
 
 	$tpl = get_markup_template('xrd_person.tpl');
 
-	$profile_url = $a->get_baseurl().'/profile/'.$r[0]['nickname'];
+	$profile_url = App::get_baseurl().'/profile/'.$r[0]['nickname'];
 
 	if ($acct)
```
Comment (Hypolite) [inline mod/xrd.php]: Standards: Could you please add brackets to this conditional statement? 😊
Standards: Could you please add a space after the `if`? 😊

File: include/salmon.php (hunk 1 above)
Comment (Quix0r) [inline include/salmon.php]: And `> 0` is fine here?

Comment (Hypolite) [inline include/salmon.php]: It is superfluous to me but you can leave it as is.

File: mod/allfriends.php (hunk 9 above)
Comment (Quix0r) [inline mod/allfriends.php]: Opps!

File: mod/events.php (hunk 19 above)
Comment (Quix0r) [inline mod/events.php]: This is next, have to go home now.

File: mod/noscrape.php (hunk 34 above)
Comment (Quix0r) [inline mod/noscrape.php]: Ah, found it.

Comment (Hypolite) [inline mod/noscrape.php]: Could you please remove your comment then? 😊

**Comment by Quix0r** [comment]
Have rewritten my PR according to feedback.

File: mod/events.php (hunk 19 above)
Comment (Hypolite) [inline mod/events.php]: What about the `if($a->argc == 1) {` ?

Comment (Quix0r) [inline mod/events.php]: Nothing found.

Comment (Hypolite) [inline mod/events.php]: Not sure I get it. Can you see line 16 of `mod/events.php` the `if` conditional with a missing space after the `if`?

Comment (Quix0r) [inline mod/events.php]: Long story short, I have fixed it now, besides some more.

**Comment by annando** [comment]
@Hypolite since we are really short before a release I second that. So @Quix0r please resolve the conflicts so that we can accept this PR.

**Comment by Quix0r** [comment]
Strangely it is loading endless ... :-( Can someone resolve it here? I have already solved it in my branch and gave edit-permission on this PR.

**Comment by Quix0r** [comment]
Ah, it was not solved by me. :-) Now it is solving nicely (because of these changes here).

File: include/like.php [hunk 45]
```diff
@@ -166,7 +166,6 @@ function do_like($item_id, $verb) {
 	$post_type = (($item['resource-id']) ? t('photo') : t('status'));
 	if ($item['object-type'] === ACTIVITY_OBJ_EVENT) {
 		$post_type = t('event');
-	}
```
Comment (rabuzarus) [inline include/like.php]: this bracket shouldn't be deleted

**Comment by rabuzarus** [comment]
there are some notes left from the merging process. See https://github.com/friendica/friendica/pull/3036/files#diff-e35a35b1057bfc56679d84a3cddff1ec

File: mod/proxy.php [hunk 46]
```diff
@@ -140,11 +140,23 @@ function proxy_init(App $a) {
 	if (!$direct_cache AND ($cachefile == '')) {
 		$r = qu("SELECT * FROM `photo` WHERE `resource-id` = '%s' LIMIT 1", $urlhash);
 		if (dbm::is_result($r)) {
+<<<<<<< HEAD
```
Comment (annando) [inline mod/proxy.php]: Please remove the merge conflicts here.

File: include/message.php [hunk 47]
```diff
@@ -149,31 +150,28 @@ function send_message($recipient=0, $body='', $subject='', $replyto=''){
 		}
 	}
 
-	if($post_id) {
+	if ($post_id) {
 		proc_run(PRIORITY_HIGH, "include/notifier.php", "mail", $post_id);
 		return intval($post_id);
-	} else {
+	}
+	else {
```
Comment (annando) [inline include/message.php]: There shouldn't be a linefeed here.

File: mod/dirfind.php [hunk 48]
```diff
@@ -85,15 +86,19 @@ function dirfind_content(&$a, $prefix = "") {
 			$perpage = 80;
 			$startrec = (($a->pager['page']) * $perpage) - $perpage;
 
-			if (get_config('system','diaspora_enabled'))
+			if (get_config('system','diaspora_enabled')) {
 				$diaspora = NETWORK_DIASPORA;
-			else
+			}
+			else {
```
Comment (annando) [inline mod/dirfind.php]: Please remove the linefeed here.

File: mod/dirfind.php [hunk 49]
```diff
@@ -85,15 +86,19 @@ function dirfind_content(&$a, $prefix = "") {
...
 				$diaspora = NETWORK_DFRN;
+			}
 
-			if (!get_config('system','ostatus_disabled'))
+			if (!get_config('system','ostatus_disabled')) {
 				$ostatus = NETWORK_OSTATUS;
-			else
+			}
+			else {
```
Comment (annando) [inline mod/dirfind.php]: Please remove the linefeed here.

File: mod/dirfind.php [hunk 50]
```diff
@@ -194,10 +200,12 @@ function dirfind_content(&$a, $prefix = "") {
 						$photo_menu = contact_photo_menu($contact[0]);
 						$details = _contact_detail_for_template($contact[0]);
 						$alt_text = $details['alt_text'];
-					} else
+					}
+					else {
```
Comment (annando) [inline mod/dirfind.php]: Please remove the linefeed here.

File: mod/follow.php [hunk 51]
```diff
@@ -170,17 +171,21 @@ function follow_post(&$a) {
...
-		if($result['message'])
+	if ($result['success'] == false) {
+		if ($result['message']) {
 			notice($result['message']);
+		}
 		goaway($return_url);
-	} elseif ($result['cid'])
-		goaway($a->get_baseurl().'/contacts/'.$result['cid']);
+	}
+	elseif ($result['cid']) {
```
Comment (annando) [inline mod/follow.php]: Please remove the linefeed here.

File: mod/home.php [hunk 52]
```diff
@@ -21,18 +22,24 @@ function home_content(&$a) {
...
-		if(file_exists('home.css')){
-			  $a->page['htmlhead'] .= '<link rel="stylesheet" type="text/css" href="'.$a->get_baseurl().'/home.css'.'" media="all" />';}
+	/// @TODO No absolute path used, maybe risky (security)
+	if (file_exists('home.html')) {
+		if (file_exists('home.css')) {
+			$a->page['htmlhead'] .= '<link rel="stylesheet" type="text/css" href="'.App::get_baseurl().'/home.css'.'" media="all" />';
+		}
 
- 		$o .= file_get_contents('home.html');}
+		$o .= file_get_contents('home.html');}
```
Comment (annando) [inline mod/home.php]: Could you move this bracket at the end of the line to a new line?

File: mod/install.php [hunk 53]
```diff
@@ -518,22 +518,24 @@ function check_smarty3(&$checks) {
...
+		}
 
 		if ($test!="ok") {
 			$status = false;
 			$help = t('Url rewrite in .htaccess is not working. Check your server configuration.');
 		}
 		check_add($checks, t('Url rewrite is working'), $status, true, $help);
-	} else {
+	}
+	else {
```
Comment (annando) [inline mod/install.php]: Please remove the linefeed here.

File: mod/install.php [hunk 54]
```diff
@@ -550,7 +552,7 @@ function check_imagik(&$checks) {
 	}
 	if ($imagick == false) {
 		check_add($checks, t('ImageMagick PHP extension is not installed'), $imagick, false, "");
-		}
+	}
 	else {
```
Comment (annando) [inline mod/install.php]: Please remove the linefeed here.

File: mod/invite.php [hunk 55]
```diff
@@ -116,11 +116,13 @@ function invite_content(&$a) {
 
 	$dirloc = get_config('system','directory');
 	if(strlen($dirloc)) {
-		if($a->config['register_policy'] == REGISTER_CLOSED)
+		if ($a->config['register_policy'] == REGISTER_CLOSED) {
 			$linktxt = sprintf( t('Visit %s for a list of public sites that you can join. Friendica members on other sites can all connect with each other, as well as with members of many other social networks.'), $dirloc . '/siteinfo');
-		elseif($a->config['register_policy'] != REGISTER_CLOSED)
-			$linktxt = sprintf( t('To accept this invitation, please visit and register at %s or any other public Friendica website.'), $a->get_baseurl())
+		}
+		elseif($a->config['register_policy'] != REGISTER_CLOSED) {
```
Comment (annando) [inline mod/invite.php]: Please remove the linefeed here.

File: mod/settings.php [hunk 56]
```diff
@@ -1164,13 +1170,14 @@ function settings_content(&$a) {
...
 	}
 
-	if(strlen(get_config('system','directory'))) {
+	if (strlen(get_config('system','directory'))) {
 		$profile_in_net_dir = replace_macros($opt_tpl,array(
 			'$field' 	=> array('profile_in_netdirectory', t('Publish your default profile in the global social directory?'), $profile['net-publish'], '', array(t('No'),t('Yes'))),
 		));
 	}
-	else
+	else {
```
Comment (annando) [inline mod/settings.php]: Please remove the linefeed here.

File: mod/starred.php [hunk 57]
```diff
@@ -34,12 +39,16 @@ function starred_init(&$a) {
...
+	if ($return_path) {
 		$rand = '_=' . time();
-		if(strpos($return_path, '?')) $rand = "&$rand";
-		else $rand = "?$rand";
-
-		goaway($a->get_baseurl() . "/" . $return_path . $rand);
+		if (strpos($return_path, '?')) {
+			$rand = "&$rand";
+		}
+		else {
```
Comment (annando) [inline mod/starred.php]: Please remove the linefeed here.

File: mod/proxy.php (hunk 46 above)
Comment (Quix0r) [inline mod/proxy.php]: I have slightly rewritten it to avoid else block (that is no needed then).

File: include/message.php (hunk 47 above)
Comment (Quix0r) [inline include/message.php]: So `} else {` is the norm?

**Comment by Quix0r** [comment]
@rabuzarus which do you mean?

**Comment by rabuzarus** [comment]
> @rabuzarus which do you mean?

* merge conflict notes in proxy.php needs to be removed
* `if ($item['object-type'] === ACTIVITY_OBJ_EVENT) {` in /include/like.php (Line 167) does have an opening bracket but no closing bracket

File: mod/proxy.php (hunk 46 above)
Comment (Hypolite) [inline mod/proxy.php]: It still is in the change log, which means it'll be merged if the PR is accepted. 🤔

File: include/like.php [hunk 58]
```diff
@@ -161,10 +164,10 @@ function do_like($item_id, $verb) {
 	$uri = item_new_uri($a->get_hostname(),$owner_uid);
 
 	$post_type = (($item['resource-id']) ? t('photo') : t('status'));
-	if($item['object-type'] === ACTIVITY_OBJ_EVENT)
+	if ($item['object-type'] === ACTIVITY_OBJ_EVENT) {
 		$post_type = t('event');
 	$objtype = (($item['resource-id']) ? ACTIVITY_OBJ_IMAGE : ACTIVITY_OBJ_NOTE );
```
Comment (Hypolite) [inline include/like.php]: As @rabuzarus noted, a closing brace is missing here.

**Comment by annando** [comment]
@Quix0r Please have a look at this file: https://github.com/friendica/friendica/blob/b3d33ee1e61782ebc6c317ee93ae32f067888796/mod/proxy.php

There are still traces of a merge conflict inside.

File: include/like.php (hunk 58 above)
Comment (Quix0r) [inline include/like.php]: Was not missing in develop branch, but here. Fixed now.

File: mod/proxy.php (hunk 46 above)
Comment (Quix0r) [inline mod/proxy.php]: Must be a committed half-merge. :-( Fixed it now by taking code from upstream/develop.

**Comment by Quix0r** [comment]
Have changed my PR again. Thank you to all reviewers: @Hypolite @annando @rabuzarus

**Comment by annando** [comment]
I will review the changes this evening. This PR is that huge that I cannot do it on the fly. Sadly there is another conflict with the PR I just accepted :-(

**Comment by Quix0r** [comment]
I have pushed fixing commits for the conflicts.

File: include/Photo.php [hunk 59]
```diff
@@ -794,8 +794,6 @@ function update_contact_avatar($avatar, $uid, $cid, $force = false) {
 
 function import_profile_photo($photo, $uid, $cid, $quit_on_error = false) {
 
-	$a = get_app();
```
Comment (Quix0r) [inline include/Photo.php]: Yay! One down again ...

**Comment by Quix0r** [comment]
I have seen that `//` has been used for multiple-line comments which is generally discouraged. Here is a general policy:

- Use `//` For single-line comments or to comment-out a single line
- Use `/* */` for multiple-line comments (see example below)

One-line comments:
```
// This is a single-line comment <-- GOOD!
// $foo = getFooFromSomething(); <-- OKAY, consider removing when really no longer used
// This is a stupid, multi-line comment <-- BAD!
// that has been wrapped to short to <-- BAD!
// to be good. <-- BAD!
/* NOISY-DEBUG: */ doLogDebug(__METHOD__, __LINE__, 'bla'); <-- _MAYBE-GOOD/BAD?_
```
Multiple-line comments:
```
/*
 * This longer comment explains in details why and what happens at below code block.
 * Please add a space around the * and keep first line and last line empty.
 */
$foo = doSomething();
```

**Comment by annando** [comment]
We are planning to make a release soon. So there will be a feature freeze. That means that after this freeze only bugfixes will be accepted for the RC branch. The RC branch will then be merged back into the develop branch (If I remember correctly). So in that time it would be good to not have touched too many files since this will create merge conflicts and so on.

**Comment by annando** [comment]
There are several places where you introduced linefeed between a bracket and an ````else```` but I really don't want to go through the whole review process again. Please don't include such many files in future pull requests anymore.

**Comment by Quix0r** [comment]
Okay.

**Comment by annando** [comment]
@Quix0r hopefully this hadn't sound too harsh? I wrote it early in the morning after one hour of code review :-)

**Comment by Quix0r** [comment]
No, it is okay. :-) I need to lookout for such things when I change so much code.

**Pull request merged**